engine/src/test/java/org/camunda/bpm/engine/test/api/mgmt/ManagementServiceGetTelemetryDataTest.java

```diff
@@ -0,0 +1,396 @@
+  }
+
+  @Test
+  public void shouldNotResetMetricsCount() {
+    // given
+    MetricsRegistry metricsRegistry = configuration.getMetricsRegistry();
+    // create command data
+    metricsRegistry.markTelemetryOccurrence(ACTIVTY_INSTANCE_START, 5);
+
+    // when
+    TelemetryData firstTelemetryData = managementService.getTelemetryData();
+    TelemetryData secondTelemetryData = managementService.getTelemetryData();
+
+    assertThat(firstTelemetryData.getProduct().getInternals().getMetrics().get(ACTIVTY_INSTANCE_START).getCount()).isEqualTo(5);
+    assertThat(secondTelemetryData.getProduct().getInternals().getMetrics().get(ACTIVTY_INSTANCE_START).getCount()).isEqualTo(5);
+  }
+
+  @Test
+  @Deployment(resources = "org/camunda/bpm/engine/test/api/oneTaskProcess.bpmn20.xml")
+  public void shouldCollectMetrics_TelemetryEnabled() {
+    // given
+    managementService.toggleTelemetry(true);
+
+    // when
+    TelemetryData telemetryDataBeforePiStart = managementService.getTelemetryData();
+    engineRule.getRuntimeService().startProcessInstanceByKey("oneTaskProcess");
+    TelemetryData telemetryDataAfterPiStart = managementService.getTelemetryData();
+
+    // then
+    assertThat(telemetryDataBeforePiStart.getProduct().getInternals().getMetrics().get(ROOT_PROCESS_INSTANCE_START).getCount()).isEqualTo(0);
+    assertThat(telemetryDataAfterPiStart.getProduct().getInternals().getMetrics().get(ROOT_PROCESS_INSTANCE_START).getCount()).isEqualTo(1);
+  }
+
+  @Test
+  @Deployment(resources = "org/camunda/bpm/engine/test/api/oneTaskProcess.bpmn20.xml")
+  public void shouldCollectMetrics_TelemetryDisabled() {
+    // given
+    managementService.toggleTelemetry(false);
+
+    // when
+    TelemetryData telemetryDataBeforePiStart = managementService.getTelemetryData();
+    engineRule.getRuntimeService().startProcessInstanceByKey("oneTaskProcess");
+    TelemetryData telemetryDataAfterPiStart = managementService.getTelemetryData();
+
+    // then
+    Metric rootPiStartMetricBeforePiStart = telemetryDataBeforePiStart.getProduct().getInternals().getMetrics().get(ROOT_PROCESS_INSTANCE_START);
+    Metric rootPiStartMetricAfterPiStart = telemetryDataAfterPiStart.getProduct().getInternals().getMetrics().get(ROOT_PROCESS_INSTANCE_START);
+
+    assertThat(rootPiStartMetricBeforePiStart.getCount()).isEqualTo(0);
+    assertThat(rootPiStartMetricAfterPiStart.getCount()).isEqualTo(1);
+  }
+
+  @Test
+  @Deployment(resources = "org/camunda/bpm/engine/test/api/oneTaskProcess.bpmn20.xml")
+  public void shouldCollectCommands_TelemetryEnabled() {
+    // given
+    managementService.toggleTelemetry(true);
+
+    // when
+    TelemetryData telemetryDataBeforePiStart = managementService.getTelemetryData();
+    // trigger Command invocation
+    managementService.isTelemetryEnabled();
+    TelemetryData telemetryDataAfterPiStart = managementService.getTelemetryData();
```

Comment:
It's a testing best practice to have one instruction in a when block, which is the functionality under test. Here, the retrieval of the first set of telemetry data and the command invocation are preparation of the test and should go into the `given` block. The code under test is line 324. This is not a critical thing and doesn't need to be changed, but is a hint for future tests that you write.